src/vm/dllimport.cpp

```diff
@@ -6034,62 +6034,78 @@ HMODULE NDirect::LoadFromNativeDllSearchDirectories(AppDomain* pDomain, LPCWSTR
 }
 
 #ifdef FEATURE_PAL
-static void DetermineLibNameVariations(const WCHAR* const** libNameVariations, int* numberOfVariations, const SString& libName, bool libNameIsRelativePath)
+static const int MaxVariationCount = 4;
+static void DetermineLibNameVariations(const WCHAR** libNameVariations, int* numberOfVariations, const SString& libName, bool libNameIsRelativePath)
 {
-    if (libNameIsRelativePath)
+    // Supported lib name variations
+    static auto NameFmt = W("%.0s%s%.0s");
```

Comment:
A nit - I would prefer not using the `auto`. We don't use it anywhere in the coreclr sources except for lambdas and some iterators. It makes it less obvious what the real type is.

Reply:
Sure. The only time I use `auto` is when the right side is explicitly typed. In this case, `W()` seemed to be the obvious string idiom so that is why I used `auto`.